### ISSUE TITLE
`manual-list-copy` rule should prefer `lst.copy()` over `list(lst)` to be more explicit

### ISSUE THREAD

**Issue by injust** [astral-sh/ruff#21593]
### Summary

On https://docs.astral.sh/ruff/rules/manual-list-copy/, the description says:

> When creating a copy of an existing list using a for-loop, prefer `list` or `list.copy` instead.

And the example suggests:

```python
original = list(range(10000))
filtered = list(original)
```

In Zed, Ruff surfaces this messaging:

> Use `list` or `list.copy` to create a copy of a list

---

`lst.copy()` should be preferred over `list(lst)` so the intention is more explicit (copying vs converting a non-list iterable to a list).

I think the example should be updated to use `list.copy()`, and either:

- Order `list.copy()` before `list()`, or
- Only suggest `list.copy()`

**Comment by amyreese** [comment]
My main concern with `data.copy()` this is that you need to know that `data` is an actual list type that implements the copy method, while `list(data)` works regardless of the type of `data` — the latter feels more "pythonic", and its performance is roughly the same[1], so IMO should be the preferred suggestion.

1:
```
amethyst@lunatone ~/scratch » uvx python -m timeit -s 'data = list(range(10000))' 'f = list(data)'
20000 loops, best of 5: 12.8 usec per loop

amethyst@lunatone ~/scratch » uvx python -m timeit -s 'data = list(range(10000))' 'f = data.copy()'
20000 loops, best of 5: 12.2 usec per loop
```

**Comment by injust** [comment]
Oh I see, I was confused by the description. The rule is about copying iterables, not lists.

> Checks for `for` loops that can be replaced by a making a copy of a list.

This should be something like:

> Checks for `for` loops that can be replaced by copying to a new list.

**Comment by injust** [comment]
The rule's description really needs to be updated. It says it's for making copies of existing lists, but it triggers for any iterable.

> Checks for `for` loops that can be replaced by a making a copy of a list.

This should be something like:

> Checks for `for` loops that can be replaced with `list()`.

And "Why is this bad?" needs to be updated to remove mentions of an "existing list".

**Comment by trilamsr** [comment]
Hey! Took a stab at this one and opened #26242.

Used your suggested framing for "What it does" (focused on the iterable→list shape rather than "existing list"), and reworked "Why is this bad?" to drop the existing-list language. Kept the violation message, example, and rule logic untouched so nothing snapshot-y moves.

Hand-wrote the PR summary per the AI policy. Happy to iterate on the wording if maintainers want a different angle.
